plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/Pom.java

```diff
@@ -89,23 +89,27 @@ public void addDependencies(List<Library> libraries) throws CoreException {
       dependencies = document.createElement("dependencies");
     }
     
-    //todo dedup
     for (Library library : libraries) {
       for (LibraryFile artifact : library.getLibraryFiles()) {
         Element dependency = document.createElement("dependency");
         MavenCoordinates coordinates = artifact.getMavenCoordinates();
         
-        Element groupId = document.createElement("groupId");
-        groupId.setTextContent(coordinates.getGroupId());
-        Element artifactId = document.createElement("artifactId");
-        artifactId.setTextContent(coordinates.getArtifactId());
-        Element version = document.createElement("version");
-        version.setTextContent(coordinates.getVersion());
-        dependency.appendChild(groupId);
-        dependency.appendChild(artifactId);
-        dependency.appendChild(version);
+        String groupId = coordinates.getGroupId();
+        String artifactId = coordinates.getArtifactId();
         
-        dependencies.appendChild(dependency);
+        if (!dependencyExists(dependencies, groupId, artifactId)) {
```

Comment:
Yes, I also thought to notify the user if the versions are different. I also would leave it to the user to figure out, but making her aware of the situation would be helpful.